src/instructlab/model/remove.py

```diff
@@ -0,0 +1,80 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+import shutil
+import sys
+
+# Third Party
+import click
+
+# First Party
+from instructlab import clickext
+from instructlab.configuration import DEFAULTS
+from instructlab.model.backends.backends import is_model_gguf, is_model_safetensors
+
+
+def remove_item(model_path: Path, model_name: str, item_type: str):
+    """Remove the specified file or directory."""
+    try:
+        if item_type == "file":
+            print(f"Removing model file: {model_name}.")
+            model_path.unlink()
+        elif item_type == "directory":
+            print(f"Removing model directory: {model_name}.")
+            shutil.rmtree(model_path)
+        print(f"Model {model_name} has been removed.")
+        sys.exit(0)
+    except OSError as e:
+        print(f"Error while trying to remove {model_name}: {e}")
+        sys.exit(1)
+
+
+@click.command(name="remove")
+@clickext.display_params
+@click.option(
+    "-m",
+    "--model",
+    required=True,
+    help="Specify the model name to remove(check with `ilab model list` first), \n"
+         "e.g. merlinite-7b-lab-Q4_K_M.gguf or instructlab/merlinite-7b-pt.",
+)
+@click.option(
+    "-y",
+    "--yes",
+    is_flag=True,
+    help="Non-interactively remove the model without asking for confirmation.",
+)
+def remove_model(model: str, yes: bool):
+    """Remove model"""
+    models_dir = Path(DEFAULTS.MODELS_DIR)
+    model_path = models_dir / model
+
+    if not model_path.exists():
+        print(f"Error: Model {model} does not exist in {models_dir}.")
+        sys.exit(1)
+
+    if model_path.is_file() and is_model_gguf(model_path):
+        item_type = "file"
+    elif model_path.is_dir() and is_model_safetensors(model_path):
+        item_type = "directory"
+    else:
+        print(f"Error: Model found at {model_path} is not a valid .gguf file or safetensors model directory.")
+        sys.exit(1)
+
+    if yes:
+        # Non-interactive deletion
+        remove_item(model_path, model, item_type)
```

Comment:
Might be better to organize this where remove_item is only called in one place.  Will help keep the code from diverging.